app/assets/javascripts/discourse/app/widgets/post-cooked.js

```diff
@@ -273,26 +273,35 @@ export default class PostCooked {
 
     $quotes.each((index, e) => {
       const $aside = $(e);
+
       if ($aside.data("post")) {
-        const quoteId = `quote-id-${$aside.data("topic")}-${$aside.data(
-          "post"
-        )}-${index}`;
-        $aside.find("blockquote").attr("id", quoteId);
-
-        this._updateQuoteElements($aside, "chevron-down");
-        const $title = $(".title", $aside);
-
-        // Unless it's a full quote, allow click to expand
-        if (!($aside.data("full") || $title.data("has-quote-controls"))) {
-          $title.on("click", (e2) => {
-            let $target = $(e2.target);
-            if ($target.closest("a").length) {
-              return true;
+        ajax(`/posts/by_number/${$aside.data("topic")}/${$aside.data("post")}`)
+          .then(() => {
+            const quoteId = `quote-id-${$aside.data("topic")}-${$aside.data(
+              "post"
+            )}-${index}`;
+            $aside.find("blockquote").attr("id", quoteId);
+
+            this._updateQuoteElements($aside, "chevron-down");
+            const $title = $(".title", $aside);
+
+            // Unless it's a full quote, allow click to expand
+            if (!($aside.data("full") || $title.data("has-quote-controls"))) {
+              $title.on("click", (e2) => {
+                let $target = $(e2.target);
+                if ($target.closest("a").length) {
+                  return true;
+                }
+                this._toggleQuote($aside);
+              });
+              $title.data("has-quote-controls", true);
+            }
+          })
+          .catch((error) => {
+            if ([403, 404].includes(error.jqXHR.status)) {
+              return $aside.find(".title").remove();
```

Comment:
We are deep in jquery land here @pmusaraj... 

I am envisioning a solution in which we rip out the `title` div completely when we don't find a topic / post match for _x quote_. It would leave us with something like so:

<img width="560" alt="Screen Shot 2022-05-11 at 11 57 21 AM" src="https://user-images.githubusercontent.com/50783505/167905575-4add4341-01d2-4f49-b100-b2dca0cb3019.png">

```
<div class="cooked"><aside class="quote no-group quote-modified" data-username="not_from_discourse_guy" data-post="1" data-topic="2259">

<blockquote>
<p>cs the following error shows in console when open topic.</p>
</blockquote>
</aside>
<p>This is not empty yay</p></div>
```